src/types/Patterns.test.ts

```diff
@@ -110,14 +37,56 @@ const validate = (data, check) => {
   })
 }
 
+const validatePattern = (
+  country: string,
+  patternType: string,
+  testData: any
+) => {
+  it(`validates ${country} ${patternType}`, () => {
+    validate(testData[country], (expr: string) =>
+      isPattern(Patterns[country][patternType], expr)
+    )
+  })
+}
+
 describe('Input Pattern Tests', () => {
-  it('validates BPNs', () => {
-    validate(TESTDATA.BPN, isBPN)
+  it('validates BPN pattern', () => {
+    validate(BPN_TEST_DATA.BPN, (expr: string) => isPattern(Patterns.BPN, expr))
   })
-  it('validates cities', () => {
-    validate(TESTDATA.CITY, isCity)
+  it('validates City pattern', () => {
+    validate(CITY_TEST_DATA.CITY, (expr: string) =>
+      isPattern(Patterns.CITY, expr)
+    )
   })
-  it('validates streets', () => {
-    validate(TESTDATA.STREET, isStreet)
+  it('validates Street pattern', () => {
+    validate(STREET_TEST_DATA.STREET, (expr: string) =>
+      isPattern(Patterns.STREET, expr)
+    )
   })
+
+  // EORI: Same Pattern for all countries
+  validatePattern('Worldwide', 'EORI', EORI_TEST_DATA)
+
+  // LEI: Same Pattern for all countries
+  validatePattern('Worldwide', 'LEI_CODE', LEI_TEST_DATA)
+
+  // VAT ID
+  validatePattern('DE', 'VAT_ID', VAT_TEST_DATA)
+  validatePattern('FR', 'VAT_ID', VAT_TEST_DATA)
+  validatePattern('MX', 'VAT_ID', VAT_TEST_DATA)
+  validatePattern('IN', 'VAT_ID', VAT_TEST_DATA)
+  validatePattern('Worldwide', 'VAT_ID', VAT_TEST_DATA)
+
+  // VIES only exists in Europe
+  validatePattern('DE', 'VIES', VIES_TEST_DATA)
+  validatePattern('FR', 'VIES', VIES_TEST_DATA)
+  validatePattern('Worldwide', 'VIES', VIES_TEST_DATA)
+
+  // Commercial Registration Number
+  validatePattern('DE', 'COMMERCIAL_REG_NUMBER', CRN_TEST_DATA)
+  validatePattern('FR', 'COMMERCIAL_REG_NUMBER', CRN_TEST_DATA)
+  validatePattern('MX', 'COMMERCIAL_REG_NUMBER', CRN_TEST_DATA)
+  validatePattern('IN', 'COMMERCIAL_REG_NUMBER', CRN_TEST_DATA)
+  validatePattern('Worldwide', 'COMMERCIAL_REG_NUMBER', CRN_TEST_DATA)
```

Comment:
this block only tests the validatePattern function and should be wrapped in a test case function.
```
it('validates patterns', () => {
  validatePattern(...)
  ...
})
```

Reply:
Test case is created inside the validate*Pattern function.

I refactored the code with describe grouping and made it more expendable for more countries.